screens/your_proposals/index.tsx

```diff
@@ -0,0 +1,296 @@
+import React, {
+	ReactElement, useCallback, useContext, useEffect, useRef, useState,
+} from 'react'
+import { useTranslation } from 'react-i18next'
+import { InfoIcon } from '@chakra-ui/icons'
+import {
+	Button,
+	Flex, HStack, Text, VStack,
+} from '@chakra-ui/react'
+import BN from 'bn.js'
+import { useRouter } from 'next/router'
+import { ApiClientsContext } from 'pages/_app'
+import Empty from 'src/components/ui/empty'
+import Heading from 'src/components/ui/heading'
+import YourApplicationCard from 'src/components/your_applications/yourApplicationCard'
+import { CHAIN_INFO } from 'src/constants/chains'
+import config from 'src/constants/config.json'
+import { useQuestbookAccount } from 'src/hooks/gasless/useQuestbookAccount'
+import NavbarLayout from 'libraries/ui/navbarLayout'
+import getAvatar from 'src/utils/avatarUtils'
+import { formatAmount, getChainIdFromResponse, getFormattedDateFromUnixTimestamp } from 'src/utils/formattingUtils'
+import { getSupportedChainIdFromSupportedNetwork } from 'src/utils/validationUtils'
+
+
+import { getUrlForIPFSHash } from 'libraries/ipfsUtils'
+import { GrantApplication, useGetMyApplicationsLazyQuery } from 'src/generated/graphql'
+
+
+const PAGE_SIZE = 20
+
+function YourProposals() {
+
+    const buildComponent = () => (
```

Comment:
why'd we need this instead of simply returning it at the end of the component?

Reply:
Want the component building on the top of the file. Just easier to navigate the file that way